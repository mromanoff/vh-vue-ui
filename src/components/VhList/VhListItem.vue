<template>
  <component
    :is="as"
    class="VhListItem"
    v-on="$listeners"
  >
    <div class="VhListItem-icon">
      <vh-icon
        v-if="icon"
        :name="icon"
      />
    </div>
    <slot />
    <div class="VhListItem-rightContent">
      <slot name="right-content" />
    </div>
  </component>
</template>

<script>
export default {
  name: 'VhListItem',
  props: {
    as: {
      type: String,
      default: 'div',
      validator: value => {
        return value.match(/(div|li|nav)/);
      },
    },
    icon: {
      type: String,
      default: null,
    },
  },
};
</script>

<style scoped>
.VhListItem {
  display: flex;
  align-items: center;
}

.VhListItem-icon {
  margin-right: var(--space--small);
}

.VhListItem-rightContent {
  margin-left: auto;
}

.VhListItem-icon:empty,
.VhListItem-rightContent:empty {
  display: none;
}
</style>
