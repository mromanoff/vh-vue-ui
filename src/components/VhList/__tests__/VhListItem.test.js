import { shallowMount } from '@vue/test-utils';
import VhListItem from '../VhListItem.vue';

describe('VhListItem.vue', () => {
  let wrapper;

  it('renders default VhListItem', () => {
    wrapper = shallowMount(VhListItem);
    expect(wrapper.props().as).toBe('div');
    expect(wrapper.props().icon).toBe(null);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhListItem li', () => {
    wrapper = shallowMount(VhListItem, {
      propsData: {
        as: 'li',
        icon: 'car',
      },
      stubs: ['vh-icon'],
    });
    expect(wrapper.props().as).toBe('li');
    expect(wrapper.props().icon).toBe('car');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhListItem nav ', () => {
    wrapper = shallowMount(VhListItem, {
      propsData: {
        as: 'nav',
        icon: 'car',
      },
      stubs: ['vh-icon'],
    });
    expect(wrapper.props().as).toBe('nav');
    expect(wrapper.props().icon).toBe('car');
    expect(wrapper).toMatchSnapshot();
  });
});
