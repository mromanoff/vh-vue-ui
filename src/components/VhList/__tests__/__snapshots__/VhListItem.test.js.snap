// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhListItem.vue renders default VhListItem 1`] = `
<div class="VhListItem">
  <div class="VhListItem-icon">
    <!---->
  </div>
  <div class="VhListItem-rightContent"></div>
</div>
`;

exports[`VhListItem.vue renders modified VhListItem li 1`] = `
<li class="VhListItem">
  <div class="VhListItem-icon">
    <vh-icon-stub name="car"></vh-icon-stub>
  </div>
  <div class="VhListItem-rightContent"></div>
</li>
`;

exports[`VhListItem.vue renders modified VhListItem nav  1`] = `
<nav class="VhListItem">
  <div class="VhListItem-icon">
    <vh-icon-stub name="car"></vh-icon-stub>
  </div>
  <div class="VhListItem-rightContent"></div>
</nav>
`;
